### ISSUE TITLE
upgrade Guava dependency to something recent

### ISSUE THREAD

**Issue by alexeyOnGitHub** [Netflix/servo#462]
Guava 19 is pretty old  https://github.com/Netflix/servo/blob/12ffcf8eb907ad9dc5425e7480ddc0584c47a1db/dependencies.properties#L9

many methods and classes were removed in Guava 20 and newer, so using "servo" library in a newer project (which may have a newer Guava like 24+) will give "method/class not found" problems

**Comment by alexeyOnGitHub** [comment]
ah, I see the project is now deprecated and no longer supported. never mind, closing this issue then

**Comment by brharrington** [comment]
Ideally servo will work fine if newer versions are pulled in. We haven't updated the dependency because it causes issues for some other uses which still require 19.

What specific errors did you see when running on a newer version?

**Comment by alexeyOnGitHub** [comment]
while older projects still use Guava 19, it may be hard to justify supporting it forever - those people would need to stay with older versions of related libraries. 

> Exception in thread "main" java.lang.NoSuchMethodError: com.google.common.base.Objects.toStringHelper(Ljava/lang/Object;)Lcom/google/common/base/Objects$ToStringHelper;
>     at com.netflix.servo.monitor.MonitorConfig.toString(MonitorConfig.java:161)
> ...
>     at com.netflix.loadbalancer.LoadBalancerBuilder.buildFixedServerListLoadBalancer(LoadBalancerBuilder.java:56)

**Comment by dmuino** [comment]
That must be a very old version of servo. I removed the usage of
Objects.toStringHelper on Oct 15 2014.

Just upgrade your version of servo and you should be good to go.

**Comment by brharrington** [comment]
> while older projects still use Guava 19, it may be hard to justify supporting it forever

As @dmuino indicated, we significantly reduced our usage of Guava several years back to minimize the number of issues we hit with compatibility changes. It would have been removed all together, but it is unfortunately used in the public API in some places.

I tried updating the guava dependency locally to the latest version, `27.1-jre`, and it compiles fine and the tests pass. So servo shouldn't be a blocker to adopting newer versions of guava assuming you are using a recent version of servo.
